File: src/main/java/ui/person/AddressTypeBean.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package ui.person;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import javax.faces.bean.ManagedBean;

import javax.faces.bean.SessionScoped;
import javax.faces.model.SelectItem;
import support.data.AddressType;

/**
 *
 * @author werpu2
 */
@ManagedBean
@SessionScoped
public class AddressTypeBean implements Serializable {

    List items = null;

    public List<SelectItem> getAddressTypes() {
        if(items == null) {
            items = new ArrayList<SelectItem>(6);
            for(AddressType adrType: AddressType.values()) {
                items.add( new SelectItem(adrType.getKey(), Integer.toString(adrType.getValue()) ));
            }
        }
        return items;
    }

}

File: src/main/scala/org/extrasapache/myfaces/codi/examples/ebean/view/person/AddressTypeBean.scala
package org.extrasapache.myfaces.codi.examples.ebean.view.person

import javax.faces.bean.{ManagedBean, SessionScoped}
import javax.faces.model.SelectItem
import reflect.BeanProperty
import org.extrasapache.myfaces.codi.examples.ebean.support.data.AddressType
import java.io.Serializable

/**
 *
 * @author Werner Punz (latest modification by $Author$)
 * @version $Revision$ $Date$
 */

@ManagedBean
@SessionScoped
class AddressTypeBean extends Serializable {
  @BeanProperty val addressTypes: java.util.List[SelectItem] = {
    AddressType.values.foldLeft(new java.util.LinkedList[SelectItem]) ((target, item) => {
      target.add(new SelectItem(item.getKey, item.getValue.toString))
      target
    })
  }
}